Lib/tarfile.py

```diff
@@ -357,14 +358,15 @@ def __init__(self, name, mode, comptype, fileobj, bufsize):
             fileobj = _StreamProxy(fileobj)
             comptype = fileobj.getcomptype()
 
-        self.name     = name or ""
-        self.mode     = mode
+        self.name = name or ""
+        self.mode = mode
         self.comptype = comptype
-        self.fileobj  = fileobj
-        self.bufsize  = bufsize
-        self.buf      = b""
-        self.pos      = 0
-        self.closed   = False
+        self.fileobj = fileobj
+        self.bufsize = bufsize
+        self.compresslevel = compresslevel
```

Comment:
No it is not needed. My first approach was to pass it as an argument to `_init_write_gz()`, with a default value of 9 (not to change the signature).  But then I got the sense that an attribute goes more naturally with the rest of the class. I can revert to the first approach if you prefer.